Support more connections

egelja opened the pull request:
Complete rework of networking and also some cleanups and logging improvements

egelja committed:
refactor: clean up main, environment variables for redis

egelja committed:
feat: web request manager

egelja committed:
ignore callbacks after `close()`

egelja committed:
improve logging

egelja committed:
increase backtrace size

egelja committed:
ignore gdb files

egelja committed:
feat: opening connections from within connections

egelja committed:
fix file closing

egelja committed:
reorganize files

egelja committed:
add base class for connections

egelja committed:
refactor: better name for function

egelja committed:
refactor: clean up includes

Commit:
refactor: safety

Commit:
fix: logging switch for libcurl

egelja committed:
fix: toggle libcurl logging with log level

egelja committed:
build: export compile commands and enable docs in dev

egelja committed:
chore: todo list

egelja committed:
feat: connections access and tracking

egelja committed:
refactor: cleanups

egelja committed:
chore: update todo list

egelja committed:
refactor: fix warning

egelja committed:
refactor: additional logging and url normalization

egelja committed:
fix: clean up warnings

egelja committed:
fix: don't segfault on dropped connection

egelja committed:
refactor: more logging

egelja committed:
chore: update todo list

echavemann commented on src/common.hpp:
```diff
@@ -9,22 +9,84 @@
 #include <utility>
 #include <vector>
 
+/***************************** Helper types **********************************/
```
(for me) why do we need this?

egelja replied:
Look at the change in processing.cpp. It’s like passing Iterable in Python.

echavemann commented on src/main.cpp:
```diff
@@ -1,15 +1,39 @@
+#include <string_view>
 #include <tuple>
 
+constexpr static std::string_view COINBASE_SUBSCRIBE_STR = R"(
```
where was this before lmao

egelja replied:
In-line lmao.

echavemann commented on src/main.cpp:
```diff
@@ -66,57 +94,63 @@ main(int argc, const char** argv)
         }
-        else {
-            prox.process_incoming_data(string_data);
+        else [[likely]] {
```
this keyword is sick wtf

egelja replied:
I know, right!? I was looking into using __builtin_likely and then found out about this.

echavemann reviewed (dismissed):
lgtm very nce

egelja committed:
refactor: comments and cleanups

egelja committed:
feat: pipe libcurl logs through quill

egelja committed:
udpate todo list

egelja committed:
better logging

egelja committed:
feat: graceful shutdown and metrics

egelja committed:
feat: graceful shutdowns pt. 2

Closes #4

egelja committed:
feat: callback time logging

egelja committed:
refactor: clean up connection class

egelja committed:
chore: update todo list

egelja committed:
refactor: Rename `RequestManager` => `Session`

egelja committed:
feat: more hexdumps

egelja committed:
refactor: cleanups, comments, and rename `manager` to `session`

Commit:
fix: use correct signal on *nix

stevenewald committed:
Added script to check if redis docker container is running, and start it otherwise

stevenewald committed:
Added script and taskfile entry to automatically pull and launch proxy

stevenewald reviewed (dismissed):
LGTM

egelja committed:
build: fix tasks on windows

egelja committed:
build: fix deps

Conan Center maintainers are being really annoying, updating
dependency versions without changing package version.

egelja commented:
Build actions passed I'm merging this
